### PR TITLE
Add command in Makefile to get a completly static binary

### PR DESCRIPTION
**Pull request by titilambert**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,11 @@ build:
 		"-X main.Version=$(VERSION)" \
 		./cmd/telegraf/telegraf.go
 
+build-for-docker:
+	GO_ENABLED=0 GOOS=linux go build -o telegraf -ldflags \
```
Comment (sparrc) [inline Makefile]: Did you mean `CGO_ENABLED=0`?

Could you also explain why this is necessary? I believe some of the linux system plugins rely on CGO and some of them may not function with this flag.

**Comment by titilambert** [comment]
@sparrc Thanks ! Yeah it's CGO_ENABLED=0 :)
My use case is: I need to use telegraf in a container to query an other container (these two containers are in the same K8S pod)

**Comment by sparrc** [comment]
@titilambert and these are linux containers that don't have access to libc? I'm curious why exactly they can't use binaries that use CGO

**Comment by titilambert** [comment]
Hum... Maybe I missed something with docker ???
If build telegraf without CGO_ENABLED=0, I get an error:

1. Here my docker file:
```
FROM scratch
COPY ./telegraf /
```
2. I build it
```
docker build -t telegraf .
```
3. I run it
```
docker run telegraf /telegraf
WARNING: Your kernel does not support memory swappiness capabilities, memory swappiness discarded.
no such file or directory
Error response from daemon: Cannot start container 1c94b30e035b517381637732067c05c0f472ed26490db5a4a4f714f6530590f4: [8] System error: no such file or directory
```

Did I miss something ?

**Comment by titilambert** [comment]
Hum... Maybe I missed something with docker ??? @sparrc 
If build telegraf without CGO_ENABLED=0, I get an error:

1 . Here my docker file:
```
  FROM scratch
  COPY ./telegraf /
```
2 . I build it
```
  docker build -t telegraf .
```
3 . I run it
```
docker run telegraf /telegraf
WARNING: Your kernel does not support memory swappiness capabilities, memory swappiness discarded.
no such file or directory
Error response from daemon: Cannot start container 1c94b30e035b517381637732067c05c0f472ed26490db5a4a4f714f6530590f4: [8] System error: no such file or directory
```

Did I miss something ?

I don't have this error with CGO_ENABLED=0

**Comment by sparrc** [comment]
@titilambert do you still need this PR merged?

**Comment by titilambert** [comment]
@sparrc yes ! I use telegraf in a "FROM strach" container. But are you agree with this feature ?

**Comment by sparrc** [comment]
yes, sorry it took me so long to get to this

**Comment by titilambert** [comment]
@sparrc Thanks for merge this one !
Will you release this static binary of telegraf for each version ?
BTW how can I get something this https://pbs.twimg.com/media/CXlgQZ7UoAE3AMZ.jpg ?

Thanks !

**Pull request closed without merging**